src/reference/asciidoc/xmpp.adoc

```diff
@@ -80,6 +80,42 @@ When started it will register a `PacketListener` that will listen for incoming X
 It forwards any received messages to the underlying adapter which will convert them to Spring Integration Messages and send them to the specified `channel`.
 It will unregister the `PacketListener` when it is stopped.
 
+Starting with _version 4.3_ the `ChatMessageListeningEndpoint` (and its `<int-xmpp:inbound-channel-adapter>`)
+supports a `org.jivesoftware.smack.filter.StanzaFilter` injection to be registered on the provided `XMPPConnection`
+together with an internal `StanzaListener` implementation.
+See their https://www.igniterealtime.org/builds/smack/docs/latest/javadoc/org/jivesoftware/smack/XMPPConnection.html#addAsyncStanzaListener%28org.jivesoftware.smack.StanzaListener,%20org.jivesoftware.smack.filter.StanzaFilter%29[JavaDocs] for more information.
+
+Also with the _version 4.3_ the `payload-expression` has been introduced for the `ChatMessageListeningEndpoint`.
+The incoming `org.jivesoftware.smack.packet.Message` represents a root object of evaluation context.
+This option is useful in case of <<xmpp-extensions>>.
+For example, for the GCM protocol we can extract the body using expression:
+
+[source,xml]
+----
+payload-expression="getExtension('google:mobile:data').json"
```

Comment:
From one side don't forget that our `<int-xmpp:inbound-channel-adapter>` is a general XMPP protocol components, and its purpose is just receive a protocol-specific data and convert it into the `Message<?>` to send to the channel.

I am against any built-in conversion in this kind of adapters. That is already a responsibility of the next `<transformer>` in the message flow.

Although I agree that you can use some converter bean from that expression.

But again: that must not be a built-in function of the Inbound Channel Adapter.